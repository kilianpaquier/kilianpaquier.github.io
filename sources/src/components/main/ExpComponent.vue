<template>
  <v-container fluid>
    <!-- Experiences title -->
    <h2
      class="font-weight-bold mb-3"
      v-text="$vuetify.lang.t(`$vuetify.experiences-title`)"
    />

    <v-expansion-panels mandatory>
      <!-- For each main experience -->
      <v-expansion-panel v-for="(exp, i) in experiences" :key="i">
        <!-- Panel header -->
        <v-expansion-panel-header class="py-0">
          <!-- List of title, subtitle, job -->
          <v-list>
            <v-list-item three-line>
              <!-- Experience avatar -->
              <v-list-item-avatar>
                <v-img
                  :src="require(`@/assets/experiences/${exp.img}`)"
                  contain
                  eager
                />
              </v-list-item-avatar>

              <!-- Experience title, date and job -->
              <v-list-item-content>
                <v-list-item-title v-text="exp.title" />
                <v-list-item-subtitle
                  v-text="$vuetify.lang.t(`$vuetify.${exp.dates}`)"
                />
                <v-list-item-subtitle
                  v-text="$vuetify.lang.t(`$vuetify.${exp.job}`)"
                />
              </v-list-item-content>
            </v-list-item>
          </v-list>
        </v-expansion-panel-header>

        <!-- Content with list of missions -->
        <v-expansion-panel-content>
          <v-divider></v-divider>

          <!-- List of missions -->
          <v-list>
            <v-list-item
              v-for="(item, j) in exp.text"
              :key="j"
              :class="{ 'mb-n2': j < exp.text.length - 1 }"
            >
              <v-list-item-content
                class="body-2"
                v-text="$vuetify.lang.t(`$vuetify.${item}`)"
              />
            </v-list-item>
          </v-list>

          <!-- Captions if needed -->
          <v-row class="text-right" no-gutters v-if="exp.captions">
            <v-col
              cols="12"
              class="caption"
              v-for="(caption, k) in exp.captions"
              :key="k"
              v-text="$vuetify.lang.t(`$vuetify.${caption}`)"
            />
          </v-row>
        </v-expansion-panel-content>
      </v-expansion-panel>

      <!-- For each old experience -->
      <v-expansion-panel>
        <!-- Header of old experiences -->
        <v-expansion-panel-header class="py-0">
          <v-list>
            <v-list-item>
              <v-list-item-content>
                <v-list-item-title
                  v-text="$vuetify.lang.t(`$vuetify.${oldExperience.title}`)"
                />
              </v-list-item-content>
            </v-list-item>
          </v-list>
        </v-expansion-panel-header>

        <!-- Content with the list of jobs -->
        <v-expansion-panel-content>
          <v-divider></v-divider>

          <!-- List of jobs -->
          <v-list>
            <v-list-item
              v-for="(item, i) in oldExperience.text"
              :key="i"
              three-line
            >
              <!-- Avatar of each job -->
              <v-list-item-avatar>
                <v-img
                  :src="require(`@/assets/experiences/${item.img}`)"
                  contain
                />
              </v-list-item-avatar>

              <!-- Content of each job -->
              <v-list-item-content>
                <v-list-item-title
                  class="body-2"
                  v-text="$vuetify.lang.t(`$vuetify.${item.title}`)"
                />
                <v-list-item-subtitle
                  v-text="$vuetify.lang.t(`$vuetify.${item.dates}`)"
                />
                <v-list-item-subtitle
                  v-text="$vuetify.lang.t(`$vuetify.${item.value}`)"
                />
              </v-list-item-content>
            </v-list-item>
          </v-list>

          <!-- Captions if needed -->
          <v-row class="text-right" no-gutters v-if="oldExperience.captions">
            <v-col
              cols="12"
              class="caption"
              v-for="(caption, k) in oldExperience.captions"
              :key="k"
              v-text="$vuetify.lang.t(`$vuetify.${caption}`)"
            />
          </v-row>
        </v-expansion-panel-content>
      </v-expansion-panel>
    </v-expansion-panels>
  </v-container>
</template>

<script lang="ts">
import { Experience, OldExperience } from "@/model/entities";
import Vue from "vue";
import Component from "vue-class-component";

@Component
export default class ExpComponent extends Vue {
  experiences: Experience[] = [
      {
          title: "Sopra Steria",
          job: "experiences.tools.title",
          dates: "experiences.tools.dates",
          img: "soprasteria.png",
          text: [
              "experiences.tools.one",
              "experiences.tools.two",
              "experiences.tools.three",
              "experiences.tools.four",
          ],
          captions: [
              "experiences.tools.cli",
              "experiences.tools.cicd",
              "experiences.tools.us",
              "experiences.tools.implementation",
          ],
      },
      {
          title: "Alten",
          job: "experiences.alten.title",
          dates: "experiences.alten.dates",
          img: "alten.png",
          text: [
              "experiences.alten.one",
              "experiences.alten.two",
              "experiences.alten.three",
          ],
      },
  ];

  oldExperience: OldExperience = {
      title: "experiences.old.title",
      text: [
          {
              title: "experiences.old.eurosys.title",
              dates: "experiences.old.eurosys.dates",
              value: "experiences.old.eurosys.value",
              img: "eurosys.png",
          },
          {
              title: "experiences.old.ddt.title",
              dates: "experiences.old.ddt.dates",
              value: "experiences.old.ddt.value",
              img: "ddt.png",
          },
          {
              title: "experiences.old.ufcv.title",
              dates: "experiences.old.ufcv.dates",
              value: "experiences.old.ufcv.value",
              img: "ufcv.png",
          },
          {
              title: "experiences.old.cgcv.title",
              dates: "experiences.old.cgcv.dates",
              value: "experiences.old.cgcv.value",
              img: "cgcv.png",
          },
          {
              title: "experiences.old.dreal.title",
              dates: "experiences.old.dreal.dates",
              value: "experiences.old.dreal.value",
              img: "dreal.png",
          },
      ],
      captions: [
          "experiences.old.ddt.caption",
          "experiences.old.ufcv.caption",
          "experiences.old.cgcv.caption",
          "experiences.old.dreal.caption",
      ],
  };
}
</script>
