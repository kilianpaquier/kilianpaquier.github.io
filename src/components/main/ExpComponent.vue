<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">
                {{ $t("experiences_title") }}
            </h2>
        </v-col>
        <v-col cols="12">
            <v-expansion-panels :model-value="[0]" mandatory>
                <v-expansion-panel v-for="(experience, e_index) in experiences" :key="'experience_' + e_index">
                    <v-expansion-panel-title hide-actions>
                        <v-list class="bg-transparent py-0">
                            <v-list-item :title="experience.title" class="py-0 ps-1">
                                <template #prepend>
                                    <v-avatar>
                                        <v-img :eager="true" :sizes="sizes" :src="experience.src" :srcset="experience.srcset" />
                                    </v-avatar>
                                </template>

                                <v-list-item-subtitle v-for="(subtitle, s_index) in experience.subtitles" :key="'experience_subtitle' + s_index" :class="{ 'mt-1': s_index > 0 }">
                                    {{ $t(subtitle) }}
                                </v-list-item-subtitle>

                                <v-row :no-gutters="true" class="mt-1">
                                    <v-col v-for="(skill, s_index) in experience.skills" :key="'experience_skill' + s_index" :class="{ 'me-2': s_index < experience.skills.length - 1 }" class="my-1" cols="auto">
                                        <v-tooltip :text="skill.name" location="bottom">
                                            <template #activator="{props}">
                                                <component :is="skill.icon" class="mx-auto no-outline" v-bind="props" width="25" />
                                            </template>
                                        </v-tooltip>
                                    </v-col>
                                </v-row>
                            </v-list-item>
                        </v-list>
                    </v-expansion-panel-title>
                    <v-expansion-panel-text>
                        <v-list class="bg-transparent text-body-2">
                            <v-list-item v-for="(mission, m_index) in experience.missions" :key="'experience_mission_' + m_index">
                                {{ $t(mission) }}
                            </v-list-item>
                        </v-list>

                        <v-row v-if="experience.captions" :no-gutters="true" class="text-end">
                            <v-col v-for="(caption, c_index) in experience.captions" :key="'experience_caption' + c_index" class="text-caption" cols="12">
                                {{ $t(caption) }}
                            </v-col>
                        </v-row>
                    </v-expansion-panel-text>
                </v-expansion-panel>

                <v-expansion-panel>
                    <v-expansion-panel-title hide-actions>
                        <span class="my-1 ms-3">
                            {{ $t("old_experiences_title") }}
                        </span>
                    </v-expansion-panel-title>

                    <v-expansion-panel-text>
                        <v-list class="bg-transparent">
                            <v-list-item v-for="(experience, e_index) in old_experiences.experiences" :key="'old_experience_' + e_index" :title="experience.title" class="ps-2" :class="{ 'mt-2': e_index > 0 }">
                                <template #prepend>
                                    <v-avatar>
                                        <v-img :sizes="sizes" :src="experience.src" :srcset="experience.srcset" />
                                    </v-avatar>
                                </template>

                                <v-list-item-subtitle v-for="(subtitle, s_index) in experience.subtitles" :key="'old_experience_subtitle' + s_index" :class="{ 'mt-1': s_index > 0 }">
                                    {{ $t(subtitle) }}
                                </v-list-item-subtitle>
                            </v-list-item>
                        </v-list>

                        <v-row :no-gutters="true" class="text-end">
                            <v-col v-for="(caption, c_index) in old_experiences.captions" :key="'old_experience_caption' + c_index" class="text-caption" cols="12">
                                {{ $t(caption) }}
                            </v-col>
                        </v-row>
                    </v-expansion-panel-text>
                </v-expansion-panel>
            </v-expansion-panels>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {experiences, old_experiences} from "@/data/experiences";

export default defineComponent({
    name: "ExpComponent",

    setup: () => ({
        sizes: "10vw",
        experiences,
        old_experiences
    })
});
</script>