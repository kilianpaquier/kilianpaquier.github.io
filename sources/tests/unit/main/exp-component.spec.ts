import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import ExpComponent from "@/components/main/ExpComponent.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("ExpComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(ExpComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create ExpComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
