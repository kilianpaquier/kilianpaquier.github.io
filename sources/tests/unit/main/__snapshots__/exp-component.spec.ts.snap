// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpComponent.vue should create ExpComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold mb-3">$vuetify.experiences-title</h2>
  <div class="v-item-group theme--light v-expansion-panels">
    <div aria-expanded="true" class="v-expansion-panel v-expansion-panel--active v-item--active"><button type="button" aria-expanded="true" class="v-expansion-panel-header py-0 v-expansion-panel-header--active">
        <div role="list" class="v-list v-sheet theme--light">
          <div tabindex="-1" role="listitem" class="v-list-item v-list-item--three-line theme--light">
            <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
              <div class="v-image v-responsive theme--light">
                <div class="v-responsive__content"></div>
              </div>
            </div>
            <div class="v-list-item__content">
              <div class="v-list-item__title">Sopra Steria</div>
              <div class="v-list-item__subtitle">$vuetify.experiences.tools.dates</div>
              <div class="v-list-item__subtitle">$vuetify.experiences.tools.title</div>
            </div>
          </div>
        </div>
        <transition-stub name="fade-transition">
          <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </transition-stub>
      </button>
      <transition-stub name="expand-transition">
        <div class="v-expansion-panel-content">
          <div class="v-expansion-panel-content__wrap">
            <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
            <div role="list" class="v-list v-sheet theme--light">
              <div tabindex="-1" role="listitem" class="v-list-item theme--light mb-n2">
                <div class="v-list-item__content body-2">$vuetify.experiences.tools.one</div>
              </div>
              <div tabindex="-1" role="listitem" class="v-list-item theme--light mb-n2">
                <div class="v-list-item__content body-2">$vuetify.experiences.tools.two</div>
              </div>
              <div tabindex="-1" role="listitem" class="v-list-item theme--light mb-n2">
                <div class="v-list-item__content body-2">$vuetify.experiences.tools.three</div>
              </div>
              <div tabindex="-1" role="listitem" class="v-list-item theme--light">
                <div class="v-list-item__content body-2">$vuetify.experiences.tools.four</div>
              </div>
            </div>
            <div class="row text-right no-gutters">
              <div class="caption col col-12">$vuetify.experiences.tools.cli</div>
              <div class="caption col col-12">$vuetify.experiences.tools.cicd</div>
              <div class="caption col col-12">$vuetify.experiences.tools.us</div>
              <div class="caption col col-12">$vuetify.experiences.tools.implementation</div>
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
    <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header py-0">
        <div role="list" class="v-list v-sheet theme--light">
          <div tabindex="-1" role="listitem" class="v-list-item v-list-item--three-line theme--light">
            <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
              <div class="v-image v-responsive theme--light">
                <div class="v-responsive__content"></div>
              </div>
            </div>
            <div class="v-list-item__content">
              <div class="v-list-item__title">Alten</div>
              <div class="v-list-item__subtitle">$vuetify.experiences.alten.dates</div>
              <div class="v-list-item__subtitle">$vuetify.experiences.alten.title</div>
            </div>
          </div>
        </div>
        <transition-stub name="fade-transition">
          <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </transition-stub>
      </button>
      <transition-stub name="expand-transition">
        <!---->
      </transition-stub>
    </div>
    <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header py-0">
        <div role="list" class="v-list v-sheet theme--light">
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="v-list-item__content">
              <div class="v-list-item__title">$vuetify.experiences.old.title</div>
            </div>
          </div>
        </div>
        <transition-stub name="fade-transition">
          <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </transition-stub>
      </button>
      <transition-stub name="expand-transition">
        <!---->
      </transition-stub>
    </div>
  </div>
</div>
`;
